src/DrupalCI/Configuration/ConfigurationManager.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace DrupalCI\Configuration;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\Common\Annotations\AnnotationRegistry;
+use Drupal\Component\Annotation\Variable;
+use Drupal\Component\Configuration\Exception\EnvironmentVariableNotSetException;
+
+class ConfigurationManager {
+
+  /**
+   * A list of all environment variables.
+   *
+   * @var array
+   */
+  protected $environmentVariables = [];
+
+  /**
+   * A list of defined variables.
+   *
+   * @var array
+   */
+  protected $variables;
+
+  /**
+   * Constructs a ConfigurationManager.
+   */
+  public function __construct() {
+    AnnotationRegistry::registerLoader('class_exists');
+    $this->discoverVariables();
+    $this->parseEnvironmentVariables();
+  }
+
+  public function test() {
+    $this->discoverVariables();
+  }
+
+  /**
+   * Load all the environment variables.
+   */
+  private function parseEnvironmentVariables() {
+    $environment_stacks = [$_ENV, $_SERVER];
```

Comment:
I only used it for developing purpose. For some reason, my environment variables didn't get injected in `$_ENV` so therefore I used `$_SERVER`. I will need to remove it when development is done. I don't use vagrant for testing this code.